zerver/tests/test_bots.py

```diff
@@ -1179,6 +1179,211 @@ def test_patch_bot_full_name_non_bot(self) -> None:
+        self.login("cordelia")
+        bot_info = {
+            "short_name": "unauthorized",
+        }
+        result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+        self.assert_json_error(result, "Insufficient permission")
+
+    def test_patch_bot_short_name_duplicate(self) -> None:
+        self.login("hamlet")
+        # Create first bot
+        bot_info = {
+            "full_name": "Bot 1",
+            "short_name": "bot1",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        # bot1 created successfully - we'll reference it via bot2's attempt to use its name
+
+        # Create second bot
+        bot_info = {
+            "full_name": "Bot 2",
+            "short_name": "bot2",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        bot2 = self.get_bot_user("bot2-bot@zulip.testserver")
+
+        # Try to change bot2's short_name to bot1's
+        bot_info = {
+            "short_name": "bot1",
+        }
+        result = self.client_patch(f"/json/bots/{bot2.id}", bot_info)
+        self.assert_json_error(result, "Email address already in use")
+
+    def test_patch_bot_short_name_invalid(self) -> None:
+        self.login("hamlet")
+        bot_info = {
+            "full_name": "The Bot of Hamlet",
+            "short_name": "hambot",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Empty short_name should fail
+        bot_info = {
+            "short_name": "",
+        }
+        result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+        self.assert_json_error(result, "Bad name or username")
+
+    def test_patch_bot_short_name_no_change(self) -> None:
+        self.login("hamlet")
+        bot_info = {
+            "full_name": "The Bot of Hamlet",
+            "short_name": "hambot",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
+        email = "hambot-bot@zulip.testserver"
+        bot = self.get_bot_user(email)
+
+        # Changing to the same short_name should be a no-op
+        bot_info = {
+            "short_name": "hambot",
+        }
+        result = self.client_patch(f"/json/bots/{bot.id}", bot_info)
+        self.assert_json_success(result)
+
+        bot.refresh_from_db()
+        self.assertEqual(bot.email, "hambot-bot@zulip.testserver")
+
+    def test_patch_bot_short_name_invalid_fake_email_domain(self) -> None:
+        self.login("hamlet")
+        # Create bot first without the invalid domain setting
+        bot_info = {
+            "full_name": "The Bot of Hamlet",
+            "short_name": "hambot",
+        }
+        result = self.client_post("/json/bots", bot_info)
+        self.assert_json_success(result)
```

Comment:
Can `create_test_bot` helper be used for all these tests instead of repeating this code again and again?